### DISCUSSION TITLE
Facing problem in egov-url-shortening service

### DISCUSSION THREAD

**Discussion by pallavii2809** [egovernments/Digit-Core#81] [Local Governance]
Since Airtel allows only two variables in the SMS template and we have used one variable to provide date and we are using other one to have link. The link contains more than 35 characters and variable can hold up to 30 characters, so we have to use external service to short this link or is there any other way to have it done? 
This is the domain name which we are using :- http://bihargrhelp.piramalswasthya.org/

**Comment by Priyanka-eGov** [comment]
We will have to use the egov-url-shortening service to shorten the URL. Currently we are upgrading the service to shorten the URL under 30 characters.
For example -
sample link - [https://qa.digit.org/property/search/PB-2023-05-05-1468]
shortened link  - [https://qa.digit.org/eus/Wx]
The above shortened url comes out to be under 30 characters. Doing this change should not cause any issues for sending message with the approved template. @Abhishek-eGov will add updated build and PR details here.

**Comment by richa1502** [comment]
Hi @Abhishek-eGov 

Please share the updated build and PR details related to URL shortening.

Thank you!

**Reply by Abhishek-eGov** [reply]
Build Link - egovio/egov-url-shortening:v1.1.2-96e8356-5

PR - https://github.com/egovernments/DIGIT-DevOps/pull/1649#issue-1709320596
        https://github.com/egovernments/DIGIT-DevOps/pull/1565#issue-1691983425
        https://github.com/egovernments/Core-Platform/pull/165#issue-1707026720
        https://github.com/egovernments/DIGIT-Dev/pull/5476#issue-1695293004

Steps:-

- In application.properties change the server.context-path to eus.
- In values.yaml change the server-context path to eus, livenessProbePath and readinessProbePath to /eus/health.
- In qa.yaml Don’t change the Service Host path. change the egov-url-shortner-endpoint to /eus/shortener.
- In any service which is using URL Shortening Service, We have to change the context path to eus.

**Comment by richa1502** [comment]
Hi Abhishek,

Thank you, so much for sharing the details.


Thanks & Regards

Richa Yadav| Project Coordinator

E: ***@***.*** ***@***.***>* | M:
+91-8755789830

W: https://walkingtree.tech

**Comment by pallavii2809** [comment]
@Abhishek-eGov @Priyanka-eGov We would like to bring to your attention that despite making the necessary changes to the egov-url-shortening service, we are still facing difficulties in receiving SMS messages. We have followed the protocol and adjusted our URL accordingly.

To provide you with the specific details, our original URL was: http://bihargrhelp.piramalswasthya.org/digit-ui/employee, and after utilizing the shortening service, we obtain the shortened URL: http://bihargrhelp.piramalswasthya.org/eus/QDm.

The challenge we currently encounter lies with our SMS provider, Airtel, which imposes a limit of 30 characters per variable. As a result, we are unable to transmit the complete shortened URL in a single SMS variable.

We kindly request your guidance and expertise in finding a solution to this issue. We seek a professional and effective approach to address this matter.

Thanks & Regards
Pallavi Rawat

**Reply by kavi-egov** [reply]
Hi Pallavi, 

Apologies for the delayed responses. We have shortened the URL from the application side as much as possible. Since the domain name you have provided itself is "bihargrhelp.piramalswasthya.org" larger than 30 character we can't reduce it any further. What can be done here is you change template in such a way so that only the last part of the url is replaced not the whole link. 

Current template will be like  "Dear user, click this link {replaceLink}"  which will be replaced.
Solution template will be  "Dear user, click this link https://bihargrhelp.piramalswasthya.org/eus/{replaceLink}" where you can replace just the variable string "QDm" which is always lesser than 30 characters. This will require code changes in every module where the template is enriched with variables, where you can only replace the last string from the short url.

I hope this helps, thanks.
